### PR TITLE
Valhalla(Tm)

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Introducing the Valhalla(Tm). 

Feeling bored after xeno trampled your body, and you don't want to spend 30 minutes watching your fellow marines getting slaughtered because they are so bad and you were the only good player?

Fear not, Valhalla(Tm) is here for YOU. Valhalla(Tm) is a beautifull resort location for ghosts, with many activities such as:

- Watching space through the windows.
- Talking to other ghosts
- Access to most of req gear to try it out
- A chem master to make all your awesome mix
- Loadouts vendor, so you can prepare for the next round
- A firing range, to test out super loadouts
- A bar
- A kitchen
- A boxing ring

And more soon!

image

////////
Clean up the vendor file, which made my life easier
Also remove Marine HQ, which annoyed in the cleaning up me and is useless
///////

LEGAL DESCLAIMER : THIS WILL NOT MESS WITH YOUR RESPAWN TIMER (it's like you stayed as a ghost) BUT THIS WILL REMOVE YOU FROM LARVA QUEUE. Valhalla(Tm) DOES NOT TAKE ANY RESPONSABILITY IF YOU END UP AT THE 23RD PLACE OF THE QUEUE

## Why It's Good For The Game

Something for ghosts to do, plus will make prep time shorter since you can prep when dead

## Changelog
:cl:
add: Valhalla(Tm) resort location. Use the join valhalla verb in OOC when you are dead to join
/:cl:

### CODE REVIEW

**Comment by novaepee** [comment]
Improv hvh in Valhalla

File: code/modules/mob/mob_verbs.dm [hunk 1]
```diff
@@ -90,6 +90,34 @@
...
+	if(valhalla_job.outfit)
+		new_fallen.delete_equipment(TRUE)
+		new_fallen.equipOutfit(valhalla_job.outfit, FALSE)
+		new_fallen.regenerate_icons()
+	valhalla_job.after_spawn(new_fallen)
+
+	log_game("[key_name(usr)] has joined Valhalla.")
+
+	mind.transfer_to(new_fallen, TRUE)
+
```
Comment (Moggram) [inline code/modules/mob/mob_verbs.dm:120]: Maybe add something like this for the admins?

to_chat(L, "<br><br><h1>[span_danger("This is not EORG! Follow all server rules.")]</h1><br><br>")

**Comment by Walarks** [comment]
The ship shape looks like something, but I just can't put my finger on it....

**Comment by ilikesaying** [comment]
the prospect of Valhalla being a sandbox...

**Comment by Moggram** [comment]
> the prospect of Valhalla being a sandbox...

This is more like being at the great hall to eat and drink.
EORG is the other side for battle.

**Comment by bowlofcereal** [comment]
add a dining area for the realism

**Comment by Luxyboo** [comment]
does this count towards ghost hours?

**Comment by jroinc** [comment]
Traditionally, the issue with adding something like this is the fact that it succs players away from the main game as aliens. Also kinda marine-biased but that's fine IMO.

Add a check only allowing Valhalla respawns if there *aren't* burrowed larba waiting? This still lets you use it, but prevents the "6 burrowed, 12 ungas chillin" issue that could potentially happen.

**Comment by BraveMole** [comment]
> add a dining area for the realism

Another mapper will take it and make it prettier / with more stuff

**Comment by BraveMole** [comment]
> does this count towards ghost hours?

no

**Comment by BraveMole** [comment]
> Traditionally, the issue with adding something like this is the fact that it succs players away from the main game as aliens. Also kinda marine-biased but that's fine IMO.
> 
> Add a check only allowing Valhalla respawns if there _aren't_ burrowed larba waiting? This still lets you use it, but prevents the "6 burrowed, 12 ungas chillin" issue that could potentially happen.

I'll be watching to see if this happens, i think people would rather play the game than going there though, but we will see

**Comment by Viniwizi** [comment]
It would be good to make so that, if you are in valhalla, you can also wait in the larvae queue.

**Comment by Misha600** [comment]
merge it right fucking now

**Comment by BraveMole** [comment]
> It would be good to make so that, if you are in valhalla, you can also wait in the larvae queue.

Yeah that's planned; needs more code though

**Comment by Misha600** [comment]
also a pvp area and a dummy human target to test gun damage on would be sick

**Comment by Kuro020** [comment]
nice ship

**Comment by MereWalker** [comment]
Bro I love this. Its public thunderdome with all the nuts and bolts. Good work!

**Comment by ATH1909** [comment]
the return of vr

**Comment by novaepee** [comment]
Boxing ring time

File: code/modules/mob/mob_verbs.dm [hunk 2]
```diff
@@ -90,6 +90,34 @@
 
 	M.key = key
 
+/mob/verb/join_valhalla()
+	set name = "Join Valhalla"
+	set category = "OOC"
+
+	if(!GLOB.valhalla_allowed)
+		to_chat(usr, span_notice("Valhalla is disabled"))
```
Comment (XSlayer300) [inline code/modules/mob/mob_verbs.dm:98]:
```suggestion
		to_chat(usr, span_notice("Valhalla is currently disabled!"))
```

File: code/modules/mob/mob_verbs.dm [hunk 3]
```diff
@@ -90,6 +90,34 @@
...
+		return
+	
+	if(stat != DEAD)
+		to_chat(usr, span_boldnotice("You must be dead to use this!"))
+		return
+
+	var/datum/job/valhalla_job = tgui_input_list(usr, "What job would you like to have in valhalla?", "Join Valhalla", GLOB.jobs_fallen_all)
```
Comment (XSlayer300) [inline code/modules/mob/mob_verbs.dm:105]:
```suggestion
	var/datum/job/valhalla_job = tgui_input_list(usr, "You are about to embark to the ghastly walls of Valhalla. What job would you like to have?", "Join Valhalla", GLOB.jobs_fallen_all)
```

File: code/modules/mob/mob_verbs.dm (hunk 1 above)
Comment (XSlayer300) [reply code/modules/mob/mob_verbs.dm:120]: I agree.

**Comment by Three333Digit** [comment]
Would be nice if you can hear dchat while in Valhalla.

**Comment by Lumipharon** [comment]
Machinery can be broken permenantly, presumably by explosions. Can you make them damage immune? Can you also provide a limitles supply of other useful materials, like phoron/larva jelly/welding fuel tanks etc? seems pretty fun though.

**Comment by XSlayer300** [comment]
Not to mention there are still (station bounced) radios around, which means the ghosts can communicate via alamo radio.

**Comment by XSlayer300** [comment]
There are errors that I mentioned in the discord (#dev-general). Fix them?

**Comment by henriquekill9576** [comment]
Adding to the fact vendors can be permanently broken, just raising their obj_integirty won't do, apparently they just get deleted by SADAR explosions for example, so might need to also fix that if we want vendors to be invincible

**Comment by Losenis** [comment]
Please make the vendors unbreakable (assuming they aren't already and server is behind the changes), joined lategame to get some loadouts going and they were broken if not outright gone.

**Comment by ThelastArchiver** [comment]
also maybe add requetorio stuff , the bike and maybe some way to spawn in dummy humans to train for surgerys, else i gotta say this is one of the best prs yet.

**Comment by bowlofcereal** [comment]
vendors are instantly destroyed by the time people use valhalla because someone thinks its funny to nade all the vendors

making them completely invulnerable would be a good idea

**Comment by novaepee** [comment]
time to make valhalla pretty

**Pull request merged**